### PR TITLE
vim-patch:8.1.2313,8.2.{2046,2769}

### PR DESCRIPTION
**Pull request by janlazo**

### CODE REVIEW

File: src/nvim/testdir/test_options.vim [hunk 1]
```diff
@@ -621,6 +621,26 @@ func Test_local_scrolloff()
...
+  redraw
+  set writedelay=10
+  let start = reltime()
+  call setline(1, repeat('x', 70))
+  redraw
+  let elapsed = reltimefloat(reltime(start))
+  set writedelay=0
+  " With 'writedelay' set should take at least 30 * 10 msec
+  " call assert_inrange(30 * 0.01, 999.0, elapsed)
+  call assert_inrange(0.01, 999.0, elapsed)
```
Comment (janlazo) [inline src/nvim/testdir/test_options.vim:639]: @bfredl Is this suppose to happen?

File: src/nvim/buffer.c [hunk 2]
```diff
@@ -1844,7 +1844,7 @@ buf_T *buflist_new(char_u *ffname_arg, char_u *sfname_arg, linenr_T lnum,
       EMSG(_("W14: Warning: List of file names overflow"));
       if (emsg_silent == 0) {
         ui_flush();
-        os_delay(3000L, true);  // make sure it is noticed
+        os_delay(3001L, true);  // make sure it is noticed
```
Comment (bfredl) [inline src/nvim/buffer.c:1847]: What is the rationale between these magic numbers and does it even apply to us?

File: src/nvim/testdir/test_options.vim (hunk 1 above)
Comment (bfredl) [reply src/nvim/testdir/test_options.vim:639]: 'writedelay' works quite differently in neovim so not necessarily.

File: src/nvim/buffer.c (hunk 2 above)
Comment (janlazo) [reply src/nvim/buffer.c:1847]: :man_shrugging: It's a vim patch. I omitted the change where `ui_delay` is logged. I don't know the alternative to Vim's `ch_log()`.

On Vim, `ui_delay()` runs either `gui_wait_for_chars()` or `mch_delay` (which runs `WaitForChar()`) so the extra milliseconds matter.

Comment (janlazo) [reply src/nvim/buffer.c:1847]: Updated the patch to use `DLOG` instead of  `ch_log`.

**Pull request merged**